build.yaml

```diff
@@ -243,25 +251,26 @@ common_cli:
     # We have a devpi cache in AWS which we will consume instead of going
     # upstream to the PyPi servers.
     # We specify that devpi caching server using -i \$PIP_INDEX_URL
-    # which requires as to include --trusted_host as we are not (yet) using
+    # which requires as to include --trusted-host as we are not (yet) using
     # SSL on our caching box.
     # The --trusted-host option is only available with pip 7.
     #
+    if [ \${PIP_INDEX_URL} ]; then
+        PIP_ADDITIONAL_OPTIONS="\${PIP_ADDITIONAL_OPTIONS} -i \${PIP_INDEX_URL} "
+    fi
+    if [ \${TRUSTED_HOST} ]; then
+        PIP_ADDITIONAL_OPTIONS="\${PIP_ADDITIONAL_OPTIONS} \
+                                --trusted-host \${TRUSTED_HOST} "
+    fi
+
+    # parse_logs() currently only works on Ubuntu or Centos builds
+    # due to a different awk implementation on OSX cloudbees servers
     if is_ubuntu || is_centos; then
-        PIP_ADDITIONAL_OPTIONS=" -i \${PIP_INDEX_URL} --trusted-host \${TRUSTED_HOST}"
         PARSE_LOGS="parse_logs "
     fi
-    if is_osx; then
-      # we can't use our pip caching layer on our OSX cloudbees boxes
-      # as we're unable run the slave_setup plugin that injects the environment
-      # variables to the slave.
-      # So we just go upstream for the python modules.
-      PIP_ADDITIONAL_OPTIONS=""
-      PARSE_LOGS=""
-    fi
 
     # using the caching-layer, install all the dependencies
-    \${PARSE_LOGS} pip install --upgrade pip \${PIP_ADDITIONAL_OPTIONS}
+    \${PARSE_LOGS} pip install --upgrade pip
     \${PARSE_LOGS} pip install . \${PIP_ADDITIONAL_OPTIONS}
```

Comment:
if our pip is <7, using the pip cache and 'trusted-host' would fail as pip<7 doesn't have --trusted-host option